.github/workflows/node.js.yml

```diff
@@ -0,0 +1,107 @@
+name: Node.js CI
+
+on:
+  push:
+    branches:
+      - jp-gha-conversion
+      - staging
+      # - master
+  pull_request:
+    branches:
+      - jp-gha-conversion
+      - staging
+      # - master
+  workflow_dispatch:
+
+jobs:
+  PreDeployTesting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 💅 Prettier check
+        run: yarn prettier:check
+
+      - name: 💨 Lint check
+        run: yarn lint
+
+      - name: 🏗️ Build app
+        run: yarn ng build --prod
+
+  Deploy:
+    if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/staging')}}
```

Comment:
I would probably say we want to support doing `workflow_dispatch` for deploys from master & staging as well. That way we can quickly force a redeploy

Reply:
I'll look up how to configure that.